### DISCUSSION TITLE
PWM duty values

### DISCUSSION THREAD

**Discussion by freedreamer82** [RavenSystem/esp-homekit-devices#2041] [General]
Hi  @RavenSystem 

why  PWM duty is not set as % 0-100 but in a range of uint16 ( 0 to 65535) ?

to set up in 50% is the half ? 65535/2 ..... and so on for other values?

thanks

**Comment by RavenSystem** [comment]
UINT16, to allow greater resolutions than percentages, and because it is the resolution used by the driver. 

Yes 50% will be 65534/2.

**Reply by freedreamer82** [reply]
ok thanks
